velox/exec/RowNumber.cpp

```diff
@@ -97,8 +101,14 @@ void RowNumber::addInput(RowVectorPtr input) {
 }
 
 void RowNumber::addSpillInput() {
+  ensureInputFits(input_);
```

Comment:
this will be called twice when invoked via addInput(), will this result in the reservation being increased twice?

Reply:
No, the `addInput` is called before `noMoreInput`, while `addInputSpill` is called after `noMoreInput`. 
The `addInputSpill` is specifically called right after the spillInputReader_ reads a batch in either the `restoreNextSpillPartition` or `getOutput`. The spillInputReader_ is null before `noMoreInput` is called, and it first becomes non-null at the end of `restoreNextSpillPartition` when the `noMoreInput` is called.